code/modules/projectiles/gun.dm

```diff
@@ -127,7 +127,7 @@
 		QDEL_NULL(chambered)
 	if(azoom)
 		QDEL_NULL(azoom)
-	UnregisterSignal(list(COMSIG_TWOHANDED_WIELD, COMSIG_TWOHANDED_UNWIELD))
+	UnregisterSignal(src, list(COMSIG_TWOHANDED_WIELD, COMSIG_TWOHANDED_UNWIELD))
```

Comment:
This line is redundant, signals get removed on destroy by the parent proc anyway